pkg/assets/create/creater.go

```diff
@@ -0,0 +1,177 @@
+package create
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+
+	"github.com/openshift/library-go/pkg/assets"
+)
+
+type CreateOptions struct {
+	// Filters allows to filter which files we will read from disk.
+	Filters []assets.FileInfoPredicate
+
+	// Context specifies time to terminate creation and report last error
+	Context context.Context
+
+	client dynamic.Interface
```

Comment:
why are some lower and some upper case?

Reply:
didn't want to expose them as they are set in ensure

Reply:
without context, this will be just place to specify the filters

Reply:
see comment below. Using this as state is ugly.

Reply:
either pass them down manually or wrap this.

Reply:
yeah, i removed client and context from the struct, cleaned up the code.